doc/source/programs/gdal_vector_sql.rst

```diff
@@ -0,0 +1,105 @@
+.. _gdal_vector_sql_subcommand:
+
+================================================================================
+"gdal vector sql" sub-command
+================================================================================
+
+.. versionadded:: 3.11
+
+.. only:: html
+
+    Apply SQL statement(s) to a dataset
+
+.. Index:: gdal vector sql
+
+Synopsis
+--------
+
+.. code-block::
+
+    Usage: gdal vector sql [OPTIONS] <INPUT> <OUTPUT> <statement>|@<filename>
+
+    Apply SQL statement(s) to a dataset.
+
+    Positional arguments:
+      -i, --input <INPUT>                                  Input vector dataset [required]
+      -o, --output <OUTPUT>                                Output vector dataset [required]
+      --sql <statement>|@<filename>                        SQL statement(s) [may be repeated] [required]
+
+    Common Options:
+      -h, --help                                           Display help message and exit
+      --version                                            Display GDAL version and exit
+      --json-usage                                         Display usage as JSON document and exit
+      --drivers                                            Display driver list as JSON document and exit
+      --config <KEY>=<VALUE>                               Configuration option [may be repeated]
+      --progress                                           Display progress bar
+
+    Options:
+      -l, --layer, --input-layer <INPUT-LAYER>             Input layer name(s) [may be repeated]
+      -f, --of, --format, --output-format <OUTPUT-FORMAT>  Output format
```

Comment:
I believe that `-l` gets skipped when `-sql` is used.

Reply:
I agree -l / --input-layer is confusing in that context. Removed for that utility